File: app/src/main/java/com/labessence/biotech/finaldilution/component/concentration/MolarConcentration.java
package com.labessence.biotech.finaldilution.component.concentration;

/**
 * Project: FinalDilution
 * Created by dawid.chmielewski on 11/11/2017.
 */

public class MolarConcentration extends Concentration {
    public MolarConcentration(double amount) {
        super(amount, ConcentrationType.MOLAR);
    }

    @Override
    public double calcDesiredMass(double volume, double molarMass) {
        return concentration * volume * molarMass / 1000;
    }

    @Override
    public double calcVolumeForDesiredMass(double mass, double molarMass) {
        return mass / molarMass / concentration * 1000;
    }
}

File: app/src/main/java/com/labessence/biotech/finaldilution/component/concentration/MolarConcentration.kt
package com.labessence.biotech.finaldilution.component.concentration

import com.labessence.biotech.finaldilution.compound.NoMolarMassException

/**
 * Project: FinalDilution
 * Created by dawid.chmielewski on 11/11/2017.
 */

class MolarConcentration(amount: Double) : Concentration(amount, ConcentrationType.MOLAR) {

    override fun calcDesiredMass(volume: Double, molarMass: Double?): Double {
        if (molarMass == null) {
            throw NoMolarMassException()
        }
        return concentration * volume * molarMass / 1000
    }

    override fun calcVolumeForDesiredMass(mass: Double, molarMass: Double?): Double {
        if (molarMass == null) {
            throw NoMolarMassException()
        }
        return mass / molarMass / concentration * 1000
    }
}
